parlai/core/agents.py

```diff
@@ -237,8 +237,11 @@ def report(self):
         num_tasks = 0
         total = 0
         for i in range(len(self.tasks)):
+            tid = self.tasks[i].getID()
             mt = self.tasks[i].report()
-            m['tasks'][self.tasks[i].getID()] = mt
+            while tid in m['tasks']:
+                tid += '_'
```

Comment:
this prevents id cloberring, so if you have e.g.
"-t babi:task1k:1,babi:task1k:2,babi:task1k:2" then your report will show the results from all three, rather than hiding the second instance of task1k:2